### PR TITLE
Allow suppressed basal to have annotations

### PR DESCRIPTION
**Pull request by darinkrauss**
@jh-bate @gniezen What it says on the tin.

### CODE REVIEW

**Comment by darinkrauss** [comment]
[[platform] Allow for annotations in suspended basal](https://trello.com/c/0VmTVdY4/256-platform-allow-for-annotations-in-suspended-basal)

File: data/types/basal/suppressed.go [hunk 1]
```diff
@@ -56,6 +58,8 @@ func (s *Suppressed) Validate(validator data.Validator, allowedDeliveryTypes []s
 			}
 		}
 	}
+
+	// validator.ValidateInterfaceArray("annotations", s.Annotations)    // TODO: Any validations? Optional? Size?
```
Comment (jh-bate) [inline data/types/basal/suppressed.go]: did you mean to leave this? why not just the todo if its a note

File: data/types/basal/suppressed_test.go [hunk 2]
```diff
@@ -101,73 +102,77 @@ var _ = Describe("Target", func() {
...
+				NewTestSuppressed(nil, nil, nil, "Weekday", nil, nil),
 				[]*service.Error{}),
 			Entry("parses object that has invalid schedule name",
 				&map[string]interface{}{"scheduleName": 0},
-				NewTestSuppressed(nil, nil, nil, nil, nil),
+				NewTestSuppressed(nil, nil, nil, nil, nil, nil),
 				[]*service.Error{
 					testData.ComposeError(service.ErrorTypeNotString(0), "/scheduleName", nil),
 				}),
+			Entry("parses object that has valid annotations",
```
Comment (jh-bate) [inline data/types/basal/suppressed_test.go]: as we have no validator at the mo is there any danger to just taking anything as being a valid annotation?

Comment (darinkrauss) [reply data/types/basal/suppressed_test.go]: Yes, technically there is. I haven't figured out a reasonable way to validate these "blobs". (The only way I can think of is to manually convert them into BSON or JSON and then calculate the string length, but that seems very "hacky", so I've left it off.)

File: data/types/basal/suppressed.go (hunk 1 above)
Comment (darinkrauss) [reply data/types/basal/suppressed.go]: It is a TODO so that if/when we have a way to validate we remember to add it here.

**Pull request merged**